src/test/coins_tests.cpp

```diff
@@ -646,26 +692,26 @@ BOOST_AUTO_TEST_CASE(anchors_test)
 
         newrt2 = tree.root();
 
-        cache.PushAnchor(tree);
-        BOOST_CHECK(cache.GetBestAnchor() == newrt2);
+        cache.PushSproutAnchor(tree);
+        BOOST_CHECK(cache.GetBestAnchor(SPROUT) == newrt2);
 
         ZCIncrementalMerkleTree test_tree;
-        BOOST_CHECK(cache.GetAnchorAt(cache.GetBestAnchor(), test_tree));
+        BOOST_CHECK(cache.GetSproutAnchorAt(cache.GetBestAnchor(SPROUT), test_tree));
 
         BOOST_CHECK(tree.root() == test_tree.root());
 
         {
             ZCIncrementalMerkleTree test_tree2;
-            cache.GetAnchorAt(newrt, test_tree2);
+            cache.GetSproutAnchorAt(newrt, test_tree2);
             
             BOOST_CHECK(test_tree2.root() == newrt);
         }
 
         {
-            cache.PopAnchor(newrt);
+            cache.PopAnchor(newrt, SPROUT);
             ZCIncrementalMerkleTree obtain_tree;
-            assert(!cache.GetAnchorAt(newrt2, obtain_tree)); // should have been popped off
-            assert(cache.GetAnchorAt(newrt, obtain_tree));
+            assert(!cache.GetSproutAnchorAt(newrt2, obtain_tree)); // should have been popped off
+            assert(cache.GetSproutAnchorAt(newrt, obtain_tree));
```

Comment:
It would be nice if these tests also did some checking to do with sapling anchors.